bindings/rust/s2n-tls-hyper/src/stream.rs

```diff
@@ -48,7 +48,16 @@ where
 {
     fn connected(&self) -> Connected {
         match self {
-            MaybeHttpsStream::Https(stream) => stream.inner().get_ref().connected(),
+            MaybeHttpsStream::Https(stream) => {
+                let connected = stream.inner().get_ref().connected();
+                let conn = stream.inner().as_ref();
+                match conn.application_protocol() {
+                    // Inform hyper to use HTTP/2 when HTTP/2 is negotiated in the ALPN.
+                    #[cfg(feature = "http2")]
+                    Some(b"h2") => connected.negotiated_h2(),
+                    _ => connected,
```

Comment:
The [hyper client panics](https://github.com/hyperium/hyper-util/blob/7a39c88e86f0677c067a6a0d5ee3cc019edf453e/src/client/legacy/client.rs#L585-L586) if HTTP/2 is attempted without the `hyper-util/http2` feature enabled. The http2 feature is checked here to allow HTTP/1 to be used instead of panicking. This case is tested in the `http2` test when negotiating HTTP/2 in the ALPN without the http2 feature enabled.